Using Tileset's optional argument causes unexpected errors/failures

I have written a test to example the failure and capture it's conditions.

The following is the pytest output which helpfully includes the my test case's source code / definition.
```
======================================================================================================== FAILURES ========================================================================================================
__________________________________________________________________________________________________ test_optional_tiles ___________________________________________________________________________________________________

    def test_optional_tiles():
        tileset = Tileset([], ALL_PENTOMINOS, [DOMINO])
        board = Rectangle(20, 20)
        problem = board.tile_with_set(tileset)
>       solution = problem.solve()

tests/test_problem.py:231: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
polyomino/problem.py:62: in solve
    self.make_problem()
polyomino/problem.py:57: in make_problem
    self.key, self.array = tiling_to_array(self.tileset, self.board)
polyomino/problem.py:25: in tiling_to_array
    key, data = tiling_to_problem(tileset, shape)
polyomino/problem.py:19: in tiling_to_problem
    vector = np.concatenate([selector, shape.bit_vector([])])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

args = ([array([ True, False, False, False, False, False, False, False, False,
       False, False, False]), 0.0],), kwargs = {}
relevant_args = [array([ True, False, False, False, False, False, False, False, False,
       False, False, False]), 0.0]

>   ???
E   ValueError: all the input arrays must have same number of dimensions, but the array at index 0 has 1 dimension(s) and the array at index 1 has 0 dimension(s)

```

Please let me know if I am just misusing the API, I'm excited to use this nifty library

I am happy to provide Python contributions to fix, but opening and discussing early. I think the pytest shows it all, but let me know if I can provide more info

Thanks for releasing and maintaining 😃

--- a/polyomino/board.py
+++ b/polyomino/board.py
@@ -27,6 +27,8 @@ class Shape(object):
                     yield translated
 
     def bit_vector(self, tile):
+        if not tile:
+            return self.bit_mask(None)
         return np.sum([self.bit_mask(sq) for sq in tile], axis=0)
 
     def bit_mask(self, sq):

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "polyomino"
-version = "0.6.1"
+version = "0.6.2"
 description = "Solve polyomino tiling problems."
 readme = "README.md"
 authors = ["Jack Grahl <jack.grahl@gmail.com>"]

--- a/tests/test_problem.py
+++ b/tests/test_problem.py
@@ -10,7 +10,7 @@ from polyomino.board import Irregular, Rectangle, Chessboard
 from polyomino.constant import DOMINO, MONOMINO, TETROMINOS
 from polyomino.constant import ALL_PENTOMINOS
 from polyomino.error import PolyominoError
-from polyomino.tileset import many
+from polyomino.tileset import many, Tileset
 
 from .strategies import polyominos
 
@@ -131,6 +131,15 @@ def test_simple_problem_biggest_first_check_array():
     assert a.shape == (65, 18)
     expected_sums = np.array([4] * 20 + [2] * 45)
     np.testing.assert_array_equal(a.sum(axis=1), expected_sums)
+
+
+def test_problem_with_no_mandatory_tiles():
+    tileset = Tileset([], ALL_PENTOMINOS, [DOMINO])
+    board = Rectangle(20, 20)
+    problem = board.tile_with_set(tileset)
+    problem.make_problem()
+    a = problem.array
+    assert a is not None
 
 
 def test_output_problem_array():

--- a/tests/test_tileset.py
+++ b/tests/test_tileset.py
@@ -1,7 +1,8 @@
 import pytest
 
 from polyomino.board import Rectangle
-from polyomino.constant import TETROMINOS
+from polyomino.constant import DOMINO, TETROMINOS
+from polyomino.constant import ALL_PENTOMINOS
 from polyomino.error import CoverWithWrongModulus, CoverWithWrongSize
 from polyomino.tileset import Tileset
 
@@ -28,3 +29,10 @@ def test_wrong_gcd():
     tileset = Tileset([], [], [tile])
     with pytest.raises(CoverWithWrongModulus):
         tileset.check(board)
+
+
+def test_optional_tiles():
+    tileset = Tileset([], ALL_PENTOMINOS, [DOMINO])
+    board = Rectangle(20, 20)
+    tileset.check(board)
+

